src/web3-utils.js

```diff
@@ -218,6 +218,14 @@ export async function getMainAccount(web3) {
   }
 }
 
+export async function getLatestBlockTimestamp(web3) {
+  const latestBlockInfo = await web3.eth.getBlock('latest')
+  if (!latestBlockInfo) {
+    return 0 // we should have a retry mechanism
```

Comment:
Now that we've moved it here, returning `0` doesn't really make sense. Can we throw an exception?